src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -433,4 +441,80 @@ export class CodeButtonComponent implements OnInit {
             return 'artemisApp.exerciseActions.cloneExerciseRepository';
         }
     }
+
+    private initTheia(profileInfo: ProfileInfo) {
+        if (profileInfo.activeProfiles?.includes(PROFILE_THEIA) && this.exercise()) {
+            // Theia requires the Build Config of the programming exercise to be set
+            // @ts-expect-error (The exercise is not undefined here)
+            this.programmingExerciseSerice.getTheiaConfig(this.exercise().id!).subscribe((theiaConfig) => {
```

Comment:
can you use an exlamation mark instead of the ts-expect-error annotation here?
```suggestion
            this.programmingExerciseSerice.getTheiaConfig(this.exercise()!.id!).subscribe((theiaConfig) => {
```

Reply:
can we avoid exlamation mark fully and add a if check before? Would prefer that :D